packages/core/react/src/WalletProvider.tsx

```diff
@@ -148,6 +148,10 @@ export function WalletProvider({
     }, [adapter]);
     const isUnloading = useRef(false);
     useEffect(() => {
+        if (walletName === SolanaMobileWalletAdapterWalletName && getIsMobile(adapters)) {
+            isUnloading.current = false;
+            return;
```

Comment:
Merely switching away from the browser to visit the mobile wallet app fires the `beforeunload` event. We do _not_ want event handlers disabled in this case, so we completely disable this hack when on mobile.